### PR TITLE
BinaryCompactObject domain: support initial Kerr horizon shape

### PR DESCRIPTION
**Pull request by nilsvu**
## Proposed changes

Allows to deform excision surfaces in the initial data to conform to spinning Kerr horizons.

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### CODE REVIEW

File: src/Domain/Creators/BinaryCompactObjectHelpers.cpp [hunk 1]
```diff
@@ -420,13 +464,7 @@ TimeDependentMapOptions<IsCylindrical>::grid_to_inertial_map(
...
       return std::make_unique<detail::gi_map<Rotation>>(rotation_map_.value());
     } else {
-      ERROR(
-          "Requesting grid to inertial map without a distorted frame and "
-          "without a Rotation or Expansion map for object "
-          << Object
-          << ". This means there are no time dependent maps. If you don't want "
-             "time dependent maps, specify 'None' for TimeDependentMapOptions. "
-             "Otherwise specify at least one time dependent map.");
+      return nullptr;
```
Comment (knelli2) [inline src/Domain/Creators/BinaryCompactObjectHelpers.cpp:467]: Why this change?

File: src/Domain/Creators/BinaryCompactObjectHelpers.hpp [hunk 2]
```diff
@@ -289,7 +301,9 @@ struct TimeDependentMapOptions {
    */
   std::unordered_map<std::string,
                      std::unique_ptr<domain::FunctionsOfTime::FunctionOfTime>>
-  create_functions_of_time(const std::unordered_map<std::string, double>&
+  create_functions_of_time(const std::optional<double>& inner_radius_A,
+                           const std::optional<double>& inner_radius_B,
```
Comment (knelli2) [inline src/Domain/Creators/BinaryCompactObjectHelpers.hpp]: I believe this information is already set from the `build_maps` function, correct? Why pass it in again here?

**Review by knelli2 — COMMENTED** [review]
Couple questions

File: src/Domain/Creators/BinaryCompactObjectHelpers.cpp (hunk 1 above)
Comment (nilsvu) [reply src/Domain/Creators/BinaryCompactObjectHelpers.cpp:467]: Expansion and rotation maps are both null, only a shape map is set. That triggered this error. So I handle the `nullptr` returned here for the grid-to-inertial maps in `create_domain`.

File: src/Domain/Creators/BinaryCompactObjectHelpers.hpp (hunk 2 above)
Comment (nilsvu) [reply src/Domain/Creators/BinaryCompactObjectHelpers.hpp]: The information isn't stored AFAICT.

Comment (knelli2) [reply src/Domain/Creators/BinaryCompactObjectHelpers.hpp]: Should we store it? I think I'd prefer that over changing this function signature since right now the signature of `create_functions_of_time` is the same as `functions_of_time` that the domain creators have

Comment (nilsvu) [reply src/Domain/Creators/BinaryCompactObjectHelpers.hpp]: Ok pushed a fixup

**Review by knelli2 — APPROVED** [review]
Looks good. You can squash

**Comment by nilsvu** [comment]
Done. Thanks for reviewing!
